app/models/theme.rb

```diff
@@ -20,6 +20,16 @@ class Theme < ActiveRecord::Base
   has_many :color_schemes
   belongs_to :remote_theme
 
+  validate :user_selectable_validation
+
+  scope :parent_themes, ->() {
+    where("id NOT IN (SELECT child_theme_id FROM child_themes)")
```

Comment:
`NOT IN` or any variations of `IN` query will be bad for performance once the result set is large. What you can do here is use a `LEFT JOIN child_themes ON themes.id = child_themes_id WHERE themes.id IS NULL`